lib/library.rb

```diff
@@ -0,0 +1,65 @@
+class Library
+    attr_reader :name, :books, :authors, :checked_out_books
+    
+    def initialize(name)
+        @name = name
+        @books = []
+        @authors = []
+        @checked_out_books = []
+        @popular_book_stats = {}
+    end
+
+    def add_author(author)
+        @authors << author
+        add_books(author)
```

Comment:
Although I'd prefer this be done dynamically, not as an instance variable, I do like that you made a helper method.